src/renderers/dom/client/ReactMount.js

```diff
@@ -724,6 +729,15 @@ var ReactMount = {
       }
     }
   },
+  /**
+   * Rather than count mounts and unmounts, we query the dom for the DOMProperty.ROOT_ATTRIBUTE_NAME. This is used to
+   * dispose of global event listeners, when the last component is removed/unmounted.
+   * @returns Returns the number of react tagged root elements.
+   * @private
+   */
+  _countReactRootElements: function() {
+    return document.querySelectorAll('[ROOT_ATTR_NAME]').length;
+  },
```

Comment:
IE8 does not have it, and it seems like a bad heuristic which internally checks every node for the attribute. Wouldn't it make more sense to just keep a counter?